api/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -37,27 +37,35 @@ const (
 	IncomingPeeringDiscovery DiscoveryType = "IncomingPeering"
 )
 
+type TrustMode string
+
+const (
+	TrustUnknown   TrustMode = ""
+	TrustTrusted   TrustMode = "Trusted"
+	TrustUntrusted TrustMode = "Untrusted"
+)
+
 // ForeignClusterSpec defines the desired state of ForeignCluster
 type ForeignClusterSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	ClusterID        string        `json:"clusterID"`
-	Namespace        string        `json:"namespace"`
-	Join             bool          `json:"join"`
-	ApiUrl           string        `json:"apiUrl"`
-	DiscoveryType    DiscoveryType `json:"discoveryType"`
-	AllowUntrustedCA bool          `json:"allowUntrustedCA"`
+	ClusterID     string        `json:"clusterID"`
+	Namespace     string        `json:"namespace"`
+	Join          bool          `json:"join"`
+	ApiUrl        string        `json:"apiUrl"`
+	DiscoveryType DiscoveryType `json:"discoveryType"`
 }
 
 // ForeignClusterStatus defines the observed state of ForeignCluster
 type ForeignClusterStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	Outgoing Outgoing `json:"outgoing,omitempty"`
-	Incoming Incoming `json:"incoming,omitempty"`
-	Ttl      int      `json:"ttl,omitempty"`
+	Outgoing Outgoing  `json:"outgoing,omitempty"`
+	Incoming Incoming  `json:"incoming,omitempty"`
+	Ttl      int       `json:"ttl,omitempty"`
+	Trusted  TrustMode `json:"trusted"`
```

Comment:
I would add the enum [annotation](https://book.kubebuilder.io/reference/markers/crd-validation.html) so that if we automatically generate the crd it reflects the struct definition